### PR TITLE
support freebsd

### LINKED ISSUE #151: Compiling fails on FreeBSD
```
% cargo build
   Compiling pprof v0.10.0 (/usr/home/dveeden/dev/pprof-rs)
error[E0425]: cannot find function `create_pipe` in this scope
  --> src/addr_validate.rs:52:35
   |
52 |         let (read_fd, write_fd) = create_pipe()?;
   |                                   ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `addr` in this scope
   --> src/profiler.rs:280:44
    |
280 |                 if profiler.is_blocklisted(addr) {
    |                                            ^^^^ not found in this scope

error[E0308]: mismatched types
   --> src/profiler.rs:205:9
    |
205 | /         unsafe {
206 | |             #[cfg(target_os = "linux")]
207 | |             {
208 | |                 let errno = *libc::__errno_location();
...   |
215 | |             }
216 | |         }
    | |_________^ expected struct `ErrnoProtector`, found `()`

Some errors have detailed explanations: E0308, E0425.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `pprof` due to 3 previous errors
```

```
% freebsd-version 
13.0-RELEASE-p11
% cargo --version
cargo 1.63.0-nightly
```

### PR DESCRIPTION
**Pull request by YangKeao**
close #151. It surprisingly works "fine" on FreeBSD with little modification.

### CODE REVIEW

**Comment by YangKeao** [comment]
@dveeden PTAL

File: src/profiler.rs [hunk 1]
```diff
@@ -319,6 +322,9 @@ extern "C" fn perf_signal_handler(
                 ))]
                 let addr = unsafe { (*ucontext).uc_mcontext.pc as usize };
 
+                #[cfg(all(target_arch = "aarch64", target_os = "freebsd"))]
+                let addr = unsafe { (*ucontext).mc_gpregs.gp_elr as usize };
```
Comment (YangKeao) [inline src/profiler.rs:326]: I'm not 100% confident about this line.

**Comment by dveeden** [comment]
The diff looks good.

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by dveeden — APPROVED** [review]

**Comment by dveeden** [comment]
Compiles on 13.2-RELEASE-p1 with 1.64.0-x86_64-unknown-freebsd

**Pull request merged**